src/freenet/client/filter/CSSTokenizerFilter.java

```diff
@@ -1814,6 +1815,21 @@ else if(HTMLelement.indexOf('#')!=-1)
 		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf("~="));
 		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf("~=")+2,
 		                    attSelection.length());
+		        } else if(attSelection.indexOf("^=")!=-1) {
+		            attSelectionParts=new String[2];
+		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf("^="));
+		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf("^=")+2,
+		                    attSelection.length());
+		        } else if(attSelection.indexOf("$=")!=-1) {
+		            attSelectionParts=new String[2];
+		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf("$="));
+		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf("$=")+2,
+		                    attSelection.length());
+		        } else if(attSelection.indexOf("*=")!=-1) {
+		            attSelectionParts=new String[2];
+		            attSelectionParts[0]=attSelection.substring(0,attSelection.indexOf("*="));
+		            attSelectionParts[1]=attSelection.substring(attSelection.indexOf("*=")+2,
+		                    attSelection.length());
```

Comment:
Whenever I see code like this I wonder if there’s a way to make it cleaner and less repetitive…

Reply:
I removed the redundancy and fixed the scoping of the attSelectionParts on the side (I’m pretty sure garbage collection does not need a reused array as help anymore).